src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionEndpoint.cs

```diff
@@ -18,12 +19,18 @@ internal class RazorCompletionEndpoint : IVSCompletionEndpoint
 {
     private readonly CompletionListProvider _completionListProvider;
     private readonly ITelemetryReporter? _telemetryReporter;
+    private readonly IOptionsMonitor<RazorLSPOptions> _optionsMonitor;
     private VSInternalClientCapabilities? _clientCapabilities;
 
-    public RazorCompletionEndpoint(CompletionListProvider completionListProvider, ITelemetryReporter? telemetryReporter)
+    public RazorCompletionEndpoint(
```

Comment:
Nit: We've been converting things to use primary construtors as we touch them. Would remove some boring boiler plate (but please keep the fields)

Reply:
I've done it in two that I touched, in other places it didn't seem like it would clean things up - e.g. in DefaultRazorConfigurationService we do the checks and creation (I could've broken it up into property initializers instead, do we want to do that?)

Reply:
If it's just null checks, then we do sometimes move them to the field initializer (ie, `private readonly CompletionListProvider _completionListProvider = completionListProvider ?? throw new ArgumentNullException(nameof(completionListProvider));`, like in https://github.com/dotnet/razor/pull/9691/files#diff-4789c06f2b474a67d3495a95cbba6f0d49450b30e6e367e2b5a4e806dc26b941R54 but I also completely agree with you that when there are checks and other validation, the advantages quickly drop off.

Totally fine with leaving DefaultRazorConfigurationService in this case.

Reply:
To make sure I understand the end goal - do we want to convert all classes to use primary constructor for consistency sake even if there isn't significant or any code cleanup?

Reply:
Personally, I think removing the constructor, and especially moving argument validation to be in-line, is significant enough code cleanup, so I would vote yes to that.